### PR TITLE
Add Refactor code

### PR DESCRIPTION
**Pull request by Jawad-Nazari**
### In this pull request, we did the following:
Applied the Single Responsibility Principle (SRP) on the app.rb file since the App class handled multiple responsibilities, such as managing books, people, rentals.
- The app.rb file refactored with a helper file book_manager.rb
- The app.rb file refactored with a helper file person_manager.rb 
- The app.rb file refactored with a helper file rental_manager.rb

- Applied the Open/Closed Principle (OCP) by introducing an BookInterface and have BookManager depend on that interface. To allow for easy extension without modifying BookManager directly.

- Applied the Dependency Inversion Principle (DIP) by modifying the App class to depend on abstractions instead of concrete implementations. So introduced the BookInterface in the App class and injected an instance of BookManager into it.

- Created an adapter class BookManagerAdapter that adapts the BookManager class to implement the interface.

- Cleared linter errors
- Updated the ReadMe file

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -95,26 +112,24 @@ def create_rental
...
     print 'ID of person: '
-    id = gets.chomp.to_i
-
+    person_id = gets.chomp.to_i
     puts 'Rentals:'
-    @rentals_list.each do |rental|
-      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
-    end
+    @rentals_list.list_all_rentals(person_id)
     @parent.show_menu
```
Comment (Ibizugbe) [inline app.rb]: Nice job so far😉. I noticed that both the main class and the app class share the responsibility of requesting and validating user inputs. To improve the code's maintainability and adhere to the Single Responsibility Principle (SRP) or SOLID principles, I would suggest you assign the responsibility of validating and requesting inputs to the `main` class😉. In implementing this, kindly take all prompts or commands to request and validate inputs to the `main` class and you can call them here where necessary😉.
By doing this, you'll achieve a clearer separation of concerns and improve the codebase's overall structure. Each class will have a single responsibility, making the code easier to understand, modify, and test.

you can take a look at [this lesson](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) for a better understanding on the SOLID principle😉.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Jawad-Nazari @Mikey1-nuel 👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅

- [X] The `main` class has been assigned the responsibility of validating and requesting input from users👍.
- [X] Classes have been broken down in different files to achieve the SOLID, KISS, DRY and YAGNI rule👍.
- [x] App works fine👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Jawad-Nazari) [reply app.rb]: Hi @Ibizugbe ,
Thanks for reviewing our code, we will come up with a solution.

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**